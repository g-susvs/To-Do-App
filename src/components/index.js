export * from './TodoAdd'
export * from './TodoCounter'
export * from './TodoList'
export * from './TodoItem'