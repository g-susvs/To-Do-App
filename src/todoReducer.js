
export const todoReducer = (initialState = [], action = {}) => {

    switch (action.type) {
        case 'ADD':
            return [action.payload, ...initialState];
            
        case 'DELETE':
            return initialState.filter(todo => todo.id != action.payload);
            
            case 'TOGGLE':
                return initialState.map(todo => {
                    if(todo.id == action.payload){
                        return {
                            ...todo,
                            done: !todo.done
                        }
                    }
                    return todo;
                });
                
        case 'EDIT':
            return initialState.map( todo => {
                if(todo.id == action.payload.id){
                    return {
                        ...todo,
                        description: action.payload.description
                    }
                }
                return todo
            });
        default:
            return initialState;
    }

}