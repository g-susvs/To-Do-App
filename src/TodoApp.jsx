import { TodoAdd, TodoCounter, TodoList,  } from "./components"

export const TodoApp = () => {
  return (
    <div className="container">
      
      <div className="todo-header">
        <img src="./src/assets/icon-logo.png" alt="logo" />
        <h1>TODO-LIST</h1>
      </div>

      <TodoAdd />
      
      <TodoCounter />

      <TodoList />

    </div>
  )
}
