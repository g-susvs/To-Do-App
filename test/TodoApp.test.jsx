import { render } from "@testing-library/react"
import { TodoApp } from "../src/TodoApp";

describe('Prueba <TodoApp />', () => {

    test('debe rederizar correctamente', () => {
        const {container} = render( <TodoApp />);

        expect(container).toMatchSnapshot();
    })

})