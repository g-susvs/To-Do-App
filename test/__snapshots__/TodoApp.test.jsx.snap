// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <TodoApp /> debe rederizar correctamente 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="todo-header"
    >
      <h1>
        TODO-APP
      </h1>
    </div>
    <div
      class="todo-add"
    >
      <input
        placeholder="Agrega una tarea"
        type="text"
        value=""
      />
      <button
        class="btn-add"
      >
        +
      </button>
    </div>
    <section
      class="todo-counter"
    >
      <div
        class="pending null"
      >
        Pendientes
        <span
          class="counter"
        >
          0
        </span>
      </div>
      <div
        class="completed null"
      >
        Completados
        <span
          class="counter"
        >
          0
        </span>
      </div>
    </section>
    <ul
      class="todo-list"
    />
  </div>
</div>
`;
